lib/models/task-dependency.js

```diff
@@ -0,0 +1,68 @@
+// Copyright 2015, EMC, Inc.
+
+'use strict';
+
+module.exports = TaskDependencyFactory;
+
+TaskDependencyFactory.$provide = 'Models.TaskDependency';
+TaskDependencyFactory.$inject = [
+    'Model',
+    'Constants'
+];
+
+function TaskDependencyFactory (Model, Constants) {
+    return Model.extend({
+        connection: 'mongo',
+        identity: 'taskdependencies',
+        attributes: {
+            domain: {
+                type: 'string',
+                defaultsTo: Constants.DefaultTaskDomain
+            },
+            graphId: {
+                type: 'string',
+                required: true
+            },
+            state: {
+                type: 'string',
+                required: true
+            },
+            evaluated: {
+                type: 'boolean',
+                defaultsTo: false
+            },
+            reachable: {
+                type: 'boolean',
+                defaultsTo: true
+            },
+            taskRunnerLease: {
+                type: 'string',
+                defaultsTo: null
+            },
+            taskRunnerHeartbeat: {
+                type: 'date',
+                defaultsTo: null
+            },
+            dependencies: {
+                type: 'json',
+                required: true
+            },
+            terminalOnStates: {
+                type: 'array',
+                defaultsTo: []
+            },
+            context: {
+                type: 'json',
+                defaultsTo: null
+            },
+            toJSON: function() {
+                // Remove waterline keys that we don't want in our dependency object
+                var obj = this.toObject();
+                delete obj.createdAt;
+                delete obj.updatedAt;
+                delete obj.id;
+                return obj;
+            }
```

Comment:
Historically we have removed the id field because it gets annoying in logging and data traversal since it's actually a BSON object, and it also can get confusing to newcomers to the code when trying to figure out the difference between an document id, an instanceId, a nodeId, etc. Also, often all we need is the instanceId for queries/internal logic and so it's unnecessary to have the id field. In this refactor, in the cases where we do want to lookup by a document ID, we are already for other reasons using the waterline mongo native method so the toJSON call gets bypassed.